src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -254,6 +385,11 @@ private sealed class RequestCompletionSource : TaskCompletionSource<HttpResponse
             public HttpRequestMessage RequestMessage { get; set; }
 
             public CurlHandler Handler { get; set; }
+
+            public CurlProxyUsePolicy ProxyPolicy { get; set; }
+
+            public IWebProxy Proxy { get; set; }
```

Comment:
Why does RequestCompletionSource needs fields for ProxyPolicy and Proxy?  Aren't those accessible via Handler._proxyPolicy and Handler._proxy?  They can't change on the handler after a request has been initiated, right?  Reducing the number of fields on an object like this reduces the amount of memory needed per request.